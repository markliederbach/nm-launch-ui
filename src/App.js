import React, { Component } from "react";
import logo from "./logo.svg";
import { observer } from "mobx-react";
import LaunchStore from "./stores/LaunchStore";
import "semantic-ui-css/semantic.min.css";
import "./css/App.css";
import { Header } from "semantic-ui-react";
import { LaunchTable } from "./components/Grid/LaunchTable";

@observer
class App extends Component {
  state = {
    store: new LaunchStore()
  };

  componentDidMount() {
    this.state.store.getLaunches();
  }

  render() {
    let resultsList = [];
    if (this.state.store.results.launches) {
    }

    return (
      <div className="App">
        <Header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Global Launch Schedule</h1>
        </Header>
        <div className="site-content">
          {this.state.store.results.launches ? (
            <LaunchTable results={this.state.store.results} />
          ) : null}
        </div>
      </div>
    );
  }
}

export default App;
