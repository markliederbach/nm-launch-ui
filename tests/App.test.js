import React from "react";
import { shallow, configure } from "enzyme";
import renderer from "react-test-renderer";
import App from "../src/App";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("App", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
    wrapper.update();
  });
  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    expect(wrapper.exists()).toBe(true);
  });
});
