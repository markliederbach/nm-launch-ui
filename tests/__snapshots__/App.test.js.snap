// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="ui header App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Global Launch Schedule
    </h1>
  </div>
  <div
    className="site-content"
  >
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="ui focus input"
      >
        <input
          name="filterValue"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Filter by name..."
          type="text"
        />
      </div>
    </form>
  </div>
</div>
`;
